diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/StringTemplateFormatRule.kt

```diff
@@ -93,12 +94,21 @@ class StringTemplateFormatRule(private val configRules: List<RulesConfig>) : Rul
                 .singleOrNull()
                 ?.treeParent
                 ?.elementType == LONG_STRING_TEMPLATE_ENTRY
-        return if (onlyOneRefExpr) {
-            !(node.treeNext.text.first().isLetterOrDigit() // checking if first letter is valid
+
+        val isArrayAccessExpression = node // this should be omitted in previous expression, used for safe warranties
+                .findAllNodesWithSpecificType(REFERENCE_EXPRESSION)
+                .singleOrNull()
+                ?.treeParent
+                ?.elementType == ARRAY_ACCESS_EXPRESSION
+
+        return if (onlyOneRefExpr && !isArrayAccessExpression) {
+            (!(node.treeNext.text.first().isLetterOrDigit() // checking if first letter is valid
                     || node.treeNext.text.startsWith("_"))
-                    || node.treeNext.elementType == CLOSING_QUOTE
-        } else {
+                    || node.treeNext.elementType == CLOSING_QUOTE)
+        } else if(!isArrayAccessExpression) {
```

Comment:
you will need to check for more corner cases -  this can be not the only problem here